common/types/string.go

```diff
@@ -60,6 +60,9 @@ func (s String) ConvertToNative(typeDesc reflect.Type) (interface{}, error) {
 			Kind: &structpb.Value_StringValue{
 				StringValue: s.Value().(string)}}, nil
 	}
+	if reflect.TypeOf(s).AssignableTo(typeDesc) {
+		return s, nil
+	}
 	if typeDesc.Kind() != reflect.String {
```

Comment:
Do wee need to check this after we check the assignableTo for the type? Or we can just throw error for unsupported type?